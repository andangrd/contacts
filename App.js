import * as React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { SplashScreen } from 'expo';
import * as Font from 'expo-font';
import { Ionicons } from '@expo/vector-icons';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import { store } from './redux';

import useLinking from './navigation/useLinking';
import StackNavigator from './navigation/StackNavigator';


export default function App(props) {
  const [isLoadingComplete, setLoadingComplete] = React.useState(false);
  const [initialNavigationState, setInitialNavigationState] = React.useState();
  const containerRef = React.useRef();
  const { getInitialState } = useLinking(containerRef);

  // Load any resources or data that we need prior to rendering the app
  React.useEffect(() => {
    async function loadResourcesAndDataAsync() {
      try {
        SplashScreen.preventAutoHide();

        // Load our initial navigation state
        setInitialNavigationState(await getInitialState());

        // Load fonts
        await Font.loadAsync({
          ...Ionicons.font,
          'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
          'roboto-regular': require('./assets/fonts/roboto-regular.ttf'),
          'alata-regular': require('./assets/fonts/alata-regular.ttf'),
        });
      } catch (e) {
        // We might want to provide this error information to an error reporting service
        console.warn(e);
      } finally {
        setLoadingComplete(true);
        SplashScreen.hide();
      }
    }

    loadResourcesAndDataAsync();
  }, []);

  if (!isLoadingComplete && !props.skipLoadingScreen) {
    return null;
  } else {
    return (
      <Provider store={ store }>
        <View style={ styles.container }>
          { Platform.OS === 'ios' && <StatusBar barStyle="default" /> }
          <NavigationContainer ref={ containerRef } initialState={ initialNavigationState }>
            <StackNavigator > </StackNavigator>
          </NavigationContainer>
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
