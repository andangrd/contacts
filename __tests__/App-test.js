import * as React from "react";

import { act, create } from "react-test-renderer";

import App from "../App";


jest.mock("expo", () => ({
  Linking: {
    makeUrl: () => "/"
  },
  SplashScreen: {
    preventAutoHide: () => 'preventAutoHide',
    hide: () => 'hide',
  },
}));
jest.mock("@react-navigation/stack", () => ({
  createStackNavigator: () => ({
    Navigator: () => "Navigator",
    Screen: () => "Screen"
  })
}));
jest.mock("../navigation/BottomTabNavigator", () => "BottomTabNavigator");
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');


describe("App", () => {
  jest.useFakeTimers();
  let component;

  // keys are date and order-of-test based, so just removed them
  const filterKeys = (state) => {
    if (state.routes) {
      return {
        ...state,
        routes: state.routes.map((route) => {
          const { key, ...others } = route
          return filterKeys(others);
        }),
      }
    }
    return state;
  };

  it(`renders the loading screen`, async () => {
    // act is used to prevent snapshot returning null
    await act(async () => {
      component = create(<App />);
    });
    expect(component).toMatchSnapshot();
  });
});