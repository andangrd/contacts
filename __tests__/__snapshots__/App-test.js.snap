// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  Navigator
</View>
`;
